modules/objdetect/test/test_charucodetection.cpp

```diff
@@ -689,6 +689,44 @@ TEST(Charuco, testmatchImagePoints)
     }
 }
 
+TEST(Charuco, testDrawDetected) {
+    vector<vector<Point2i>> detected_int = {{Point(20, 20), Point(80, 20), Point(80, 80), Point2f(20, 80)}};
+    vector<vector<Point2f>> detected_float = {{Point2f(20.f, 20.f), Point2f(80.f, 20.f), Point2f(80.f, 80.f), Point2f(20.f, 80.f)}};
+    vector<vector<Point2d>> detected_double = {{Point2d(20., 20.), Point2d(80., 20.), Point2d(80., 80.), Point2d(20., 80.)}};
+    Point center = (detected_int[0][0] + detected_int[0][1] + detected_int[0][2] + detected_int[0][3]) / 4;
+
+    vector<vector<Point>> contours;
+    Point detectedCenter;
+    Moments m;
+    Mat img;
+
+    img = Mat::zeros(100, 100, CV_8UC1);
+    ASSERT_NO_THROW(aruco::drawDetectedMarkers(img, detected_float, noArray(), Scalar(255, 255, 255)));
+    findContours(img, contours, RETR_EXTERNAL, CHAIN_APPROX_SIMPLE);
+    ASSERT_EQ(contours.size(), 1ull);
+    m = moments(contours[0]);
+    detectedCenter = Point(cvRound(m.m10/m.m00), cvRound(m.m01/m.m00));
+    ASSERT_EQ(detectedCenter, center);
+
+    img = Mat::zeros(100, 100, CV_8UC1);
+    ASSERT_NO_THROW(aruco::drawDetectedCornersCharuco(img, detected_float[0], noArray(), Scalar(255, 255, 255)));
+    findContours(img, contours, RETR_EXTERNAL, CHAIN_APPROX_SIMPLE);
+    ASSERT_EQ(contours.size(), 4ull);
+    for (size_t i = 0ull; i < 4ull; i++) {
+        m = moments(contours[i]);
+        detectedCenter = Point(cvRound(m.m10/m.m00), cvRound(m.m01/m.m00));
+        ASSERT_TRUE(find(detected_int[0].begin(), detected_int[0].end(), detectedCenter) != detected_int[0].end());
```

Comment:
`drawDetectedCornersCharuco` is called for _float, but checked for _int. Also _double version is initialized, but not used.

Reply:
detected_int and detected_float are equal, using int will help to avoid rounding errors
the "_double" version will be added